File: array_compare/array_compare.py
#!/usr/bin/env python3

import os
import sys


class ArrayCompare():
    def __init__(self, fname):
        self.fname = fname
        os.path.exists(self.fname)

    def compare(self):
        try:
            with os.popen('./%s' % self.fname) as f:
                data = f.read()
                return data
        except Exception as e:
            print(e)
            input('Please input any key for exitting:')
            sys.exit(1)


if __name__ == '__main__':
    if len(sys.argv) != 2:
        print("Input parameters for one, please check your parameter!")
        sys.exit(1)

    fname = sys.argv[1]

    ac = ArrayCompare(fname)
    data = ac.compare()
    data = data.split()
    key = data[0]
    for d in data:
        if key == d:
            print(d, end=' ')
        else:
            print('\033[0;31m%s\033[0m' % d, end=' ')
    print()

File: array_compare/array_compare.c
/*
 * Comparing if all values of one array is the same.
 */

#include <stdio.h>

int main(int argc, char **argv)
{
	int key;
	int data[20];
	unsigned int i;

	/* Initial array */
	for (i = 0; i < sizeof(data) / sizeof(int); i++) {
		if ((i == 6) || (i == 8) || (i == 9) || (i == 15) || (i == 18))
			data[i] = i;
		else
			data[i] = 1;
	}

	/* Compare array's value */
	key = data[0];
	for (i = 0; i < sizeof(data) / sizeof(int); i++) {
		if (key == data[i])
			printf("%d ", data[i]);
		else
			printf("%d(%d) ", data[i], i);
	}
	printf("\n");

	return 0;
}
